### PR TITLE
Placate the JDK 9 compiler

### PR DESCRIPTION
**Pull request by jglick**
[JDK-8075793](https://bugs.openjdk.java.net/browse/JDK-8075793)

Useful preparation, and allows newer versions of NetBeans to edit Jenkins sources despite [NB bug 268452](https://netbeans.org/bugzilla/show_bug.cgi?id=268452).

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/util/Iterators.java [hunk 1]
```diff
@@ -314,12 +315,13 @@ public void remove() {
      * <p>
      * That is, this creates {A,B,C,D} from {A,B},{C,D}.
      */
+    @SafeVarargs
     public static <T> Iterable<T> sequence( final Iterable<? extends T>... iterables ) {
         return new Iterable<T>() {
             public Iterator<T> iterator() {
-                return new FlattenIterator<T,Iterable<? extends T>>(Arrays.asList(iterables)) {
+                return new FlattenIterator<T,Iterable<? extends T>>(ImmutableList.copyOf(iterables)) {
```
Comment (jglick) [inline core/src/main/java/hudson/util/Iterators.java]: Not really related but figured I would solve some other compiler warnings while I was here.

File: core/src/main/java/hudson/util/Iterators.java [hunk 2]
```diff
@@ -350,8 +352,9 @@ protected boolean filter(T t) {
         };
     }
 
+    @SafeVarargs
```
Comment (jglick) [inline core/src/main/java/hudson/util/Iterators.java]: Ditto. Internally `concat` calls `ImmutableList.copyOf`.

**Comment by oleg-nenashev** [comment]
:bee:

**Review by daniel-beck — APPROVED** [review]

**Comment by jglick** [comment]
@reviewbybees done

**Pull request merged**